internal/internal_event_handlers.go

```diff
@@ -560,6 +567,9 @@ func validateVersion(changeID string, version, minSupported, maxSupported Versio
 func (wc *workflowEnvironmentImpl) GetVersion(changeID string, minSupported, maxSupported Version) Version {
 	if version, ok := wc.changeVersions[changeID]; ok {
 		validateVersion(changeID, version, minSupported, maxSupported)
+		if wc.isReplay {
```

Comment:
if ok then this is not the first call to GetVersion with this changeID. Why do we need to Upsert in this case?

Reply:
Because when replay, marker will be processed first then code will enter this block, it needs to do same upsert as first GetVersion call. Otherwise replay will error.